tensorboard/uploader/server_info.py

```diff
@@ -152,6 +154,32 @@ def allowed_plugins(server_info):
         return frozenset((scalars_metadata.PLUGIN_NAME,))
 
 
+def max_blob_size(server_info):
+    """Determines the maximum allowed size for blob uploads.
+
+    This pulls from the `max_blob_size` on the `server_info` when that
+    submessage is set, else falls back to a default.
+
+    Note that the RPC endpoint also enforces a limit, which should be kept in
+    sync with the one provided by server_info.  Here the purpose is to provide
+    a better error message and to avoid attempting the upload if it is expected
+    to fail anyway.
+
+    Args:
+      server_info: A `server_info_pb2.ServerInfoResponse` message.
+
+    Returns:
+      A integer giving the maximum size allowed for blob uploads, in bytes.
+    """
+    if server_info.HasField("max_blob_size"):
+        return server_info.plugin_control.max_blob_size
```

Comment:
This dereferences `max_blob_size` from `plugin_control`, but the proto
actually has it on the root response. Please add tests (as for sibling
function `allowed_plugins`) that would catch this.

Reply:
Whoopsiedoodle.  Fixed and added tests.